### ISSUE TITLE
Add Expr.quantile and Series.quantile

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#203]
https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.quantile.html#polars.Expr.quantile

I'm _slightly_ worried that the default interpolation method might change in Polars at some point, so we should probably not put any default for it in Narwhals and force users to specify it explicitly

**Comment by FBruzzesi** [comment]
I can take this. Just to brainstorm ideas, we can keep a regression test comparing with polars default. That would give us an hint if the polars default changes in the future. WDYT?

**Comment by MarcoGorelli** [comment]
true, but if the polars default changes, then we'll have to change the default here too. ideally, the narwhals api should be as stable as possible (even if polars/pandas aren't)

if this is aimed at library devs rather than end users, then "forcing" them to specify `method=` in return for guaranteed stability is probably worth it

**Comment by FBruzzesi** [comment]
> true, but if the polars default changes, then we'll have to change the default here too

That's where a test can help

> if this is aimed at library devs rather than end users, then "forcing" them to specify method= in return for guaranteed stability is probably worth it

I have mixed feeling on this. On one side, you are absolutely right, on the other hand, if the api wants to reflect polars behaviour flawlessly, I think the same default of polars is required 🤔

**Comment by MarcoGorelli** [comment]
> reflect polars behaviour flawlessly

kinda - the objective is to make a strict subset of the Polars API, and to me, making a keyword-argument required fits the definition of "strict subset"

So if somebody writes their codebase using Narwhals, and then they want to switch to just supporting Polars, then `nw -> pl` should do the trick (barring `nw.from_native`-style calls). However, the other way round isn't guaranteed

**Comment by FBruzzesi** [comment]
Adjusted in #207
